### PR TITLE
TP: 7225, Comment: Adds BackToTop as a Dough component

### PR DESCRIPTION
**Pull request by davidtrussler**
This adds a component to Dough that adds a link for smaller devices to return the user to the top of the page. This is activated when the user scrolls down the content by a certain amount.

### CODE REVIEW

File: assets/js/components/BackToTop.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        this.$btt_link.css({
+          'position': 'absolute',
+          'top': 0 - this.linkHeight,
+          'bottom': 'auto'
+        });
+      } else {
+        this.$btt_link.css({
+          'position': 'fixed',
+          'bottom': 0,
+          'top': 'auto'
```
Comment (slaywell) [inline assets/js/components/BackToTop.js]: Might be tidier to do this by adding a class instead, and putting the styles in a CSS file? Everything except the `'top': 0 - this.linkHeight,` anyway?

**Comment by slaywell** [comment]
A few things on this:

**Comment by slaywell** [comment]
A few thoughts on this:

- We should have at least minimal styles for the button within Dough in order to make it work without Frontend
- Can you add a description to this PR?
- What did we decide with unit tests, are they going to come later or can you add a basic one now?

File: assets/js/components/BackToTop.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  /**
+  * @param {Promise} initialised
+  */
+  BackToTop.prototype.init = function(initialised) {
+    this._setUpBTTLink();
+    this._initialisedSuccess(initialised);
+  };
+
+  return BackToTop;
+});
```
Comment (slaywell) [inline assets/js/components/BackToTop.js]: Needs a new line!

File: assets/stylesheets/components/common/_back_to_top.scss [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+.btt-link--shown {
+  visibility: visible;
+}
+
+.btt-link__arrow {
+  width: 20px;
+  height: 12.1572px;
+  margin-left: 11px;
+}
```
Comment (slaywell) [inline assets/stylesheets/components/common/_back_to_top.scss]: This should have a new line too

File: assets/stylesheets/components/common/_back_to_top.scss [hunk 4]
```diff
@@ -0,0 +1,33 @@
+.btt-link {
```
Comment (slaywell) [inline assets/stylesheets/components/common/_back_to_top.scss]: Can we make this class consistent with the file name for BEM? I.e. call both file and class `btt-link`, or perhaps even `back-to-top`?

It's not obvious to a new pair of eyes what `btt-link` means.

**Comment by johnplayer1982** [comment]
Tiny thing but can you squash your commits once you've made Matt's change above

**Comment by slaywell** [comment]
👍

**Comment by johnplayer1982** [comment]
:+1:

**Pull request merged**